client/state/plugins/installed/selectors.js

```diff
@@ -66,65 +66,131 @@ export function requestPluginsError( state ) {
 	return state.plugins.installed.requestError;
 }
 
-function getPluginsSelector( state, siteIds, pluginFilter ) {
-	let pluginList = reduce(
-		siteIds,
-		( memo, siteId ) => {
-			if ( isRequesting( state, siteId ) ) {
-				return memo;
-			}
+const getSiteIdsThatHavePlugins = createSelector(
+	( state ) => {
+		return Object.keys( state.plugins.installed.plugins ).map( ( siteId ) => Number( siteId ) );
+	},
+	( state ) => [ state.plugins.installed.plugins ]
+);
+
+/**
+ * The server returns plugins store at state.plugins.installed.plugins are indexed by site, which means
+ * that the information for a plugin may be spread across multiple site objects. This selector transforms
+ * that structure into one indexed by the plugin slugs and memoizes that structure.
+ */
+export const getAllPluginsIndexedByPluginSlug = createSelector(
```

Comment:
This selector serves to perform the computation previously being done in `getPlugins` once, memoize it, and expose it in a way that it can be accessed in O(1) time if you have a plugin slug.